pkg/provider/azure_loadbalancer.go

```diff
@@ -1924,49 +2149,93 @@ func (az *Cloud) reconcileBackendPools(clusterName string, service *v1.Service,
 			klog.V(10).Infof("reconcileLoadBalancer for service (%s)(%t): lb backendpool - found wanted backendpool. not adding anything", serviceName, wantLb)
 			foundBackendPool = true
 
-			var backendIPConfigurationsToBeDeleted []network.InterfaceIPConfiguration
-			if bp.BackendAddressPoolPropertiesFormat != nil && bp.BackendIPConfigurations != nil {
-				for _, ipConf := range *bp.BackendIPConfigurations {
-					ipConfID := to.String(ipConf.ID)
-					nodeName, _, err := az.VMSet.GetNodeNameByIPConfigurationID(ipConfID)
-					if err != nil && !errors.Is(err, cloudprovider.InstanceNotFound) {
-						return false, false, err
-					}
+			var (
+				backendIPConfigurationsToBeDeleted []network.InterfaceIPConfiguration
+				nodeIPAddressesToBeDeleted         []string
+			)
+
+			if bp.BackendAddressPoolPropertiesFormat != nil {
+				if strings.EqualFold(az.LoadBalancerBackendPoolConfigurationType, consts.LoadBalancerBackendPoolConfigurationTypeVMNIC) &&
+					bp.BackendIPConfigurations != nil {
```

Comment:
is it possible to put the backend pool implementations behind the VMSet interface?

Reply:
created a new interface LoadBalancerBackendPool